fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailabilitySpec.groovy

```diff
@@ -0,0 +1,241 @@
+
+    ConstraintTrackingAvailability testAvailability
+    DataSourceConstraint testConstraint
+    ColumnMaskingAvailability maskedAvailability
+
+    Dimension filter1
+    Dimension filter2
+    Dimension other1
+    Dimension other2
+
+    String filter1Name
+    String filter2Name
+    String other1Name
+    String other2Name
+
+    def setup() {
+        filter1Name = "filter1"
+        filter1 = Mock(Dimension)
+        filter1.getApiName() >> filter1Name
+
+        filter2Name = "filter2"
+        filter2 = Mock(Dimension)
+        filter2.getApiName() >> filter2Name
+
+        other1Name = "other1"
+        other1 = Mock(Dimension)
+        other1.getApiName() >> other1Name
+
+        other2Name = "other2"
+        other2 = Mock(Dimension)
+        other2.getApiName() >> other2Name
+
+
+        testConstraint = new BaseDataSourceConstraint(
+                [other1, filter1] as Set,
+                [other2, filter2] as Set,
+                [] as Set,
+                [] as Set,
+                new ApiFilters()
+        )
+        testAvailability = new ConstraintTrackingAvailability(filter1Name, filter2Name)
+        maskedAvailability = new ColumnMaskingAvailability(testAvailability, [filter1Name, filter2Name] as Set)
+    }
+
+    def "all constrained availability methods have filtered dimensions removed from constraint"() {
+        when:
+        testAvailability.clearConstraintDimensions()
+        maskedAvailability.getDataSourceNames(testConstraint)
+
+        then:
+        testAvailability.constraintDimensions == [other1, other2] as Set
+
+        when:
+        testAvailability.clearConstraintDimensions()
+        maskedAvailability.getAvailableIntervals(testConstraint)
+
+        then:
+        testAvailability.constraintDimensions == [other1, other2] as Set
+
+        when:
+        testAvailability.clearConstraintDimensions()
+        maskedAvailability.getExpectedStartDate(testConstraint)
+
+        then:
+        testAvailability.constraintDimensions == [other1, other2] as Set
+
+        when:
+        testAvailability.clearConstraintDimensions()
+        maskedAvailability.getExpectedEndDate(testConstraint)
+
+        then:
+        testAvailability.constraintDimensions == [other1, other2] as Set
+    }
+    def "Test column masking when constrained by  physical constraint"() {
+        setup:
+        String missingDimensionPhysicalName = "missing_dimension"
+        String missingDimensionApiName = "missingDimension"
+
+        AvailabilityTest target = new AvailabilityTest()
+        //build map of druid physical columns and their available time intervals
+        target.physicalAvailabity = [
+                column_1 : new SimplifiedIntervalList([ new Interval("2020-01-01/2020-03-01")]),
+                column_2 : new SimplifiedIntervalList([ new Interval("2020-02-01/2020-03-01")]),
+                missingDimensionPhysicalName : new SimplifiedIntervalList(),
```

Comment:
needs parenthesis around `missingDimensionPhysicalName`. If there are no parenthesis groovy interprets the key value as a string literal and not a variable reference.